src/components/Comments.jsx

```diff
@@ -1,28 +1,67 @@
 import { useEffect, useState } from "react"
 import axios from "axios";
 import CommentCard from "./CommentCard";
+import { getCommentsByArticleId, postNewComment } from "../api/apiFunctions";
+
 
 function Comments({article_id}) {
 
     const [comments, setComments] = useState([])
+    const [newComment, setNewComment] = useState("")
+    const [loggedInUser] = useState("jessjelly")
+    const [commentData, setCommentData] = useState({})
+    const [commentPending, setCommentPending] = useState(false)
+    const [refresh, setRefresh] = useState(0)
+
+    function handleSubmitComment(e) {
+        setCommentPending(true)
+        e.preventDefault()
+        postNewComment(article_id, commentData)
+        .then((response)=>{
+            setCommentPending(false)
+            setNewComment("")
+            setRefresh(Math.random())
+        })
+        .catch((error)=>{
+            setCommentPending(false)
+        })
+    }
+
+    useEffect(()=>{
```

Comment:
rather than using a useEffect here, I would suggest directly creating the comment object to post within the handleSubmit - currently, this useEffect will run every time the user types into the input field which is unnecessary. you'll be able to see this happening if you add a console.log to this useEffect :)